tpl/tplimpl/template_funcs_test.go

```diff
@@ -1021,6 +1023,52 @@ func TestIntersect(t *testing.T) {
 	}
 }
 
+func TestUnion(t *testing.T) {
+	t.Parallel()
+	for i, this := range []struct {
+		sequence1 interface{}
+		sequence2 interface{}
+		expect    interface{}
+	}{
+		{[]string{"a", "b", "c", "c"}, []string{"a", "b", "b"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b"}, []string{"a", "b", "c"}, []string{"a", "b", "c"}},
+		{[]string{"a", "b", "c"}, []string{"d", "e"}, []string{"a", "b", "c", "d", "e"}},
+		{[]string{}, []string{}, []string{}},
+		{[]string{"a", "b"}, nil, make([]interface{}, 0)},
```

Comment:
I agree that nil == empty set in this context. But What is `{1,2,3} ∪ {}`? What you `expect` is the `intersect`, not the `union`.